lib/active_model/serializer.rb

```diff
@@ -80,24 +80,36 @@ def self.serializers_cache
 
     # @api private
     # Find a serializer from a class and caches the lookup.
-    # Preferentially returns:
-    #   1. class name appended with "Serializer"
-    #   2. try again with superclass, if present
-    #   3. nil
     def self.get_serializer_for(klass)
       return nil unless config.serializer_lookup_enabled
       serializers_cache.fetch_or_store(klass) do
-        # NOTE(beauby): When we drop 1.9.3 support we can lazify the map for perfs.
-        serializer_class = serializer_lookup_chain_for(klass).map(&:safe_constantize).find { |x| x && x < ActiveModel::Serializer }
-
-        if serializer_class
-          serializer_class
-        elsif klass.superclass
-          get_serializer_for(klass.superclass)
-        end
+        get_serializer_class(klass)
+      end
+    end
+
+    # @api private
+    # Preferentially returns:
+    #   1. class name appended with "Serializer"
+    #   2. try again with superclass, if present
+    #   3. (optional) runs `on_serializer_not_found` policy if serializer is not found
+    #   4. nil
+    def self.get_serializer_class(klass)
+      # NOTE(beauby): When we drop 1.9.3 support we can lazify the map for perfs.
+      serializer_class = serializer_lookup_chain_for(klass).map(&:safe_constantize).find { |x| x && x < ActiveModel::Serializer }
+
+      if serializer_class
+        serializer_class
+      elsif klass.superclass && !excluded_base_class?(klass.superclass)
```

Comment:
see, where I'm getting caught up on this, is that I'm reading this as:

"if the objec i'm trying to serialize has a superclass, and that superclass is neither ActiveRecord::Base, BasicObject, nor Object, then get the serializer for that ....."

And here is the problem (in my looking things over too quickly -- you're getting the serializer for the superclass, not klass -- this makes sense now -- sorry about the confusion).

So, I guess for clarity, the condition should be extracted to a method -- maybe

`is_superclass_allowed_to_be_serializable?(klass)

but then, why _don't_ we want those superclasses serializable? I know it's weird otherwise, but what if someone wanted to try to write a general serializer for `Object`? would that be super weird? idk.